src/lib/nav/NavigationRailItem.svelte

```diff
@@ -0,0 +1,206 @@
+<script lang="ts">
+  import type { HTMLButtonAttributes, HTMLAnchorAttributes } from "svelte/elements";
+  import type { IconifyIcon } from "@iconify/types";
+
+  import Layer from "$lib/misc/Layer.svelte";
+  import Icon from "$lib/misc/_icon.svelte";
+
+  let {
+    label,
+    icon,
+    active = false,
+    badge,
+    onclick,
+    ...props
+  }: {
+    label: string;
+    icon: IconifyIcon;
+    active?: boolean;
+    badge?: string | boolean;
+    onclick?: (e: MouseEvent) => void;
+  } & (({ href: string } & HTMLAnchorAttributes) | HTMLButtonAttributes) = $props();
+</script>
+
+{#if 'href' in props}
+  <a class="m3-container" role="menuitem" class:active {...props}>
+    {@render item()}
+  </a>
+{:else}
+  <button class="m3-container" type="button" role="menuitem" class:active {onclick} {...props}>
+    {@render item()}
+  </button>
+{/if}
+
+{#snippet item()}
+  <Layer />
+
+  <div class="icon">
+    <Icon {icon} />
+
+    {#if typeof badge === "string" || badge === true}
+      <span class="badge m3-font-label-small" class:small={badge === true}>
+        {typeof badge === "string" ? badge : ""}
+      </span>
+    {/if}
+  </div>
+
+  {label}
+{/snippet}
+
+<style>
+  .m3-container {
+    background: none;
+    padding: 0;
+    border: none;
+    cursor: pointer;
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    cursor: pointer;
+    gap: 4px;
+    user-select: none;
+    color: rgb(var(--m3-scheme-secondary));
+    width: fit-content;
+    position: relative;
+    font-family: var(--m3-font-body, var(--m3-font));
+    animation: none !important;
+  }
+
+  .icon {
+    position: relative;
+  }
+
+  .icon > :global(svg) {
+    width: 24px;
+    height: 24px;
+    color: rgb(var(--m3-scheme-on-secondary-container));
+    position: relative;
+  }
+
+  .icon > .badge {
+    border-radius: var(--m3-util-rounding-full);
+    background: rgb(var(--m3-scheme-error));
+    color: rgb(var(--m3-scheme-on-error));
+    width: 16px;
+    height: 16px;
+    position: absolute;
+    top: 0;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+  }
+
+  .icon > .badge.small {
+    width: 6px;
+    height: 6px;
+    font-size: 0px;
+  }
+
+  @container items (width <= 56px) {
+    .m3-container {
+      font-size: var(--m3-font-label-medium-size, 0.75rem);
+      line-height: var(--m3-font-label-medium-height, 1.333);
+      letter-spacing: var(--m3-font-label-medium-tracking, 0.031rem);
+      font-weight: var(--m3-font-label-medium-weight, 500);
+    }
+    
+    .m3-container.active > .icon {
+      background: rgb(var(--m3-scheme-secondary-container));
+    }
+
+    .m3-container:not(.active):hover > .icon {
+      background: color-mix(in srgb-linear, currentColor 8%, transparent);
```

Comment:
```suggestion
      background: color-mix(in oklab, currentColor 8%, transparent);
```

As per [here](https://github.com/KTibow/m3-svelte/commit/ea33ed78b4958600e402a31f6eb9f72bba1632ee)